### PR TITLE
Add modern gen utility options (rename, relearn) to party/box menus

### PR DESCRIPTION
**Pull request by notjagan**
## Description

Adds (toggleable) options to the party/box menus for the following features available in modern generations:

- Rename: change mon's nickname (party and box)
- Relearn (TODO): relearn previous moves (party)

TODO:

- [x] Add move renamer option
- [x] Add move relearner option
- [ ] Make party/box menus scrollable (#3918)
  - [ ] Alternative for submenu

## Images

TODO

## **Discord contact info**

notnotjagan

### CODE REVIEW

File: src/party_menu.c [hunk 1]
```diff
@@ -196,6 +198,9 @@ struct PartyMenuBoxInfoRects
     u8 descTextHeight;
 };
 
+// Allocate based on number of optional actions enabled
+#define ACTION_COUNT    8 + P_CAN_RENAME_FROM_PARTY
```
Comment (notjagan) [inline src/party_menu.c:202]: don't love this but i couldn't think of a better way

File: src/party_menu.c [hunk 2]
```diff
@@ -2985,6 +3003,42 @@ static void CB2_ReturnToPartyMenuFromSummaryScreen(void)
...
+static void CursorCb_Relearn(u8 taskId)
+{
+    PlaySE(SE_SELECT);
+    gSpecialVar_0x8004 = gPartyMenu.slotId;
+    sPartyMenuInternal->exitCallback = CB2_UpdateRelearnerScreen;
+    Task_ClosePartyMenu(taskId);
+    TeachMoveRelearnerMove();
+}
+
+static void CB2_UpdateRelearnerScreen(void)
```
Comment (notjagan) [inline src/party_menu.c:3036]: it feels wrong to do this but i couldn't figure out how to make it work otherwise -- would appreciate advice

**Comment by kittenchilly** [comment]
I believe we wanted to add the move relearning stuff by pressing a button in the summary screen, instead of an option in the party menu, since that's how it is in SV and we have limited space for party menu options

**Comment by notjagan** [comment]
> I believe we wanted to add the move relearning stuff by pressing a button in the summary screen, instead of an option in the party menu, since that's how it is in SV and we have limited space for party menu options

people in #expansion-dev were discussing that it would work in the party menu options if either the menu was made to scroll or a submenu was added (of which i was planning to add both alternatives as options)

**Comment by Bassoonian** [comment]
Superseded by #5513

**Pull request closed without merging**